### PR TITLE
community: Azure Search Vector Store is missing Access Token Authentication

### PR DESCRIPTION
**Pull request by levalencia**
Added Azure Search Access Token Authentication instead of API KEY auth.
Fixes Issue: https://github.com/langchain-ai/langchain/issues/24263
Dependencies: None
Twitter: @levalencia

@baskaryan

Could you please review? First time creating a PR that fixes some code.

### CODE REVIEW

**Comment by levalencia** [comment]
can someone take a look at this, I cant remove the new dependency azure.identiy
@baskaryan

**Comment by levalencia** [comment]
I am tagging some contributors to this file as I need some help to continue:
1. Azure.Identity is a dependency that I need to be able to create an AccessToken to be able to use the token as a mean of Authentication instead of the API KEY, I noticed this is implemented in OpenAI but not in AzureSearch.py

I tried to fix it but I get errors with the depency, if I add the dependency using Poetry, then the dependecy check fails, if I dont, then I get different errors.

@thedavgar @mattgotteiner 

Your help is greatly appreciated.

File: libs/community/langchain_community/vectorstores/azuresearch.py [hunk 1]
```diff
@@ -23,6 +24,7 @@
 )
 
 import numpy as np
+from azure.core.credentials import AccessToken, TokenCredential
```
Comment (mattgotteiner) [inline libs/community/langchain_community/vectorstores/azuresearch.py:27]: This needs to be an optional import next to the other Azure imports

**Comment by mattgotteiner** [comment]
> I am tagging some contributors to this file as I need some help to continue:
> 
> 1. Azure.Identity is a dependency that I need to be able to create an AccessToken to be able to use the token as a mean of Authentication instead of the API KEY, I noticed this is implemented in OpenAI but not in AzureSearch.py
> 
> I tried to fix it but I get errors with the depency, if I add the dependency using Poetry, then the dependecy check fails, if I dont, then I get different errors.
> 
> @thedavgar @mattgotteiner
> 
> Your help is greatly appreciated.

https://python.langchain.com/v0.2/docs/contributing/code/setup/#working-with-optional-dependencies

**Comment by levalencia** [comment]
extended_testing_deps.txt already has the azure-identity

> > I am tagging some contributors to this file as I need some help to continue:
> > 
> > 1. Azure.Identity is a dependency that I need to be able to create an AccessToken to be able to use the token as a mean of Authentication instead of the API KEY, I noticed this is implemented in OpenAI but not in AzureSearch.py
> > 
> > I tried to fix it but I get errors with the depency, if I add the dependency using Poetry, then the dependecy check fails, if I dont, then I get different errors.
> > @thedavgar @mattgotteiner
> > Your help is greatly appreciated.
> 
> https://python.langchain.com/v0.2/docs/contributing/code/setup/#working-with-optional-dependencies

I added azure-core to extended_testing_deps.txt, azure-identity was already there.

```
azure-core>=1.29.7,<2
azure-identity>=1.15.0,<2
```

However I have this error:
https://github.com/langchain-ai/langchain/actions/runs/9977236809/job/27571413449?pr=24330#step:6:48

> langchain_community/vectorstores/azuresearch.py:27: in <module>
>     from azure.core.credentials import AccessToken, TokenCredential
> E   ModuleNotFoundError: No module named 'azure'

that can be solved using poetry add, but if I do then I get a test dependency error, because it should be optional.

or shouldnt I use poetry add?

**Review by levalencia — COMMENTED** [review]
@mattgotteiner  canb you approve?

**Comment by levalencia** [comment]
@baskaryan can you merge or something else is needed?

**Comment by levalencia** [comment]
@baskaryan  I have tried all communication channels for this PR to be approved, in the meantime changes have happened on the community repo and several versions have been released, almost 30 days waiting time seems a lot, or maybe I am reaching the wrong person?

**Comment by efriis** [comment]
Hey @levalencia ! Sorry for the delay. You did nothing wrong, and we're working on publishing better expectations for getting community PRs reviewed and how to escalate issues, based on [VSCode's issue triaging](https://github.com/microsoft/vscode/wiki/Issues-Triaging).

Thank you for your contribution!

File: libs/community/langchain_community/vectorstores/azuresearch.py [hunk 2]
```diff
@@ -71,17 +71,6 @@
 MAX_UPLOAD_BATCH_SIZE = 1000
 
 
-class BearerTokenCredential(TokenCredential):
```
Comment (clairebehue) [inline libs/community/langchain_community/vectorstores/azuresearch.py]: @levalencia any reason you replaced this `BearerTokenCredential` class ? `TokenCredential` doesn't seem instanciable (see stacktrace: https://github.com/langchain-ai/langchain/issues/26216 )
 i was planning to reintroduce this wrapper class : proposed fix https://github.com/langchain-ai/langchain/pull/26995)

**Pull request merged**